main.tf

```diff
@@ -18,10 +18,24 @@ resource "azurerm_mssql_server" "vaconfigured" {
   version                      = "12.0"
   administrator_login          = "4dm1n157r470r"
   administrator_login_password = "4-v3ry-53cr37-p455w0rd"
-  minimum_tls_version          = "1.1"
+  minimum_tls_version          = "1.2"
 
   transparent_data_encryption_key_vault_key_id = azurerm_key_vault_key.vaconfigured.id
 
+
+  extended_auditing_policy {
+    storage_endpoint                        = "https://accurics.com"
+    retention_in_days                       = 120
+    storage_account_access_key              = "some-key"
+    storage_account_access_key_is_secondary = true
+  }
+
+  extended_auditing_policy {
+    storage_account_access_key              = "some-key"
+    storage_account_access_key_is_secondary = true
+    storage_endpoint                        = "https://accurics.com"
+    retention_in_days                       = 120
+  }
 }
```

Comment:
**Tenable.cs remediation**
 
**[HIGH]: Ensure that the Expiration Date is set for all Keys in RBAC Key Vaults**
 
**Vulnerability**
Description:

Ensure that all Keys in Role Based Access Control (RBAC) Azure Key Vaults have an expiration time set.

Rationale:

Azure Key Vault enables users to store and use cryptographic keys within the Microsoft Azure environment. The 'exp' (expiration time) attribute identifies the expiration time on or after which the key MUST NOT be used for encryption of new data, wrapping of new keys, and signing. By default, keys never expire. It is thus recommended that keys be rotated in the key vault and set an explicit expiration time for all keys to help enforce the key rotation. This ensures that the keys cannot be used beyond their assigned lifetimes.

Keys cannot be used beyond their assigned expiration times respectively. Keys need to be rotated periodically wherever they are used.
 
**Remediation**
**From Azure Portal**

1. Go to 'Key vaults'
2. For each Key vault, click on 'Keys'.
3. Under the 'Settings' section, Make sure 'Enabled?' is set to Yes
4. Set an appropriate 'EXPIRATION DATE' on all keys.

**From Azure CLI**

Update the 'EXPIRATION DATE' for the key using below command.


az keyvault key set-attributes --name <keyName> --vault-name <vaultName> --expires Y-m-d'T'H:M:S'Z'


**Note:**
In order to access expiration time on all keys in Azure Key Vault using Microsoft API requires "List" Key permission.

To provide required access follow below steps, 

**For Key Vaults using "Key Vault Access Policy"**

1. Go to 'Key vaults'
2. For each Key vault, click on 'Access Policy'.
3. Add access policy with 'Key permission' as 'List'

**For Key Vaults using "Role Based Access Control"**

Assign the role of _"Key Vault Secrets Officer"_ to the appropriate user

**From Powershell**


Set-AzKeyVaultKeyAttribute -VaultName <Vault Name> -Name <Key Name> -Expires <DateTime>

source: tenablecs-